projects/js-packages/connection/hooks/use-product-checkout-workflow/index.jsx

```diff
@@ -67,6 +79,8 @@ export default function useProductCheckoutWorkflow( {
 	const run = event => {
 		event && event.preventDefault();
 		setCheckoutStarted( true );
+		isUserConnected && initializeAnalytics();
+		analytics.tracks.recordEvent( productSlug + '_purchase_button_click', {} );
```

Comment:
we should not create programmatically the events name.

> They should be hardcoded so they are easily searchable in the code base.

Reply:
Also, there is no guarantee that the purchase-process triggers by clicking a button. `run` is a function that could be called in different ways.